documentation/source/Checkpoints.md

```diff
@@ -0,0 +1,282 @@
+> - dataset_params - containing configuration files for instantiating different datasets and dataloaders
+> - training_hyperparams - containing configuration files holding hyper-parameters for specific recipes
+
+And now, let's take a look at the default parameters in `checkpoint_params`:
+
+```yaml
+
+load_backbone: False # whether to load only the backbone part of the checkpoint
+checkpoint_path: # checkpoint path that is located in super_gradients/checkpoints
+strict_load: True # key matching strictness for loading checkpoint's weights
+pretrained_weights: # a string describing the dataset of the pre-trained weights (for example, "imagenent").
+
+```
+
+And note the above parameters are used to start the training with different weights (fine-tuning etc.) - they are passed to model.get() in the underlying flow of `Trainer.train_from_config(...)`:
+```python
+
+    @classmethod
+    def train_from_config(cls, cfg: Union[DictConfig, dict]) -> Tuple[nn.Module, Tuple]:
+        ...
+
+        # BUILD NETWORK
+        model = models.get(
+            ...
+            strict_load=cfg.checkpoint_params.strict_load,
+            pretrained_weights=cfg.checkpoint_params.pretrained_weights,
+            checkpoint_path=cfg.checkpoint_params.checkpoint_path,
+            load_backbone=cfg.checkpoint_params.load_backbone,
+        )
+
+        # INSTANTIATE DATA LOADERS
+
+        train_dataloader = ...
+        val_dataloader = ...
+        
+        ...
+
+        # TRAIN
+        res = trainer.train(...)
+... 
+```
+
+## Resuming Training
+
+In SG, we separate the logic of resuming training from loading model weights. Therefore continuing training is controlled by two arguments, passed through `training_params`: `resume` and `resume_path`:
+```yaml
+...
+resume: False # whether to continue training from ckpt with the same experiment name.
+resume_path: # Explicit checkpoint path (.pth file) to resume training.
+...
+```
+
+Setting `resume=True` will take the training related state_dicts from `/PATH/TO/MY_CKPT_ROOT_DIR/MY_EXPERIMENT_NAME/ckpt_latest.pth`.
+Stating explicitly a `resume_path` will continue training from an explicit checkpoint.
+
+In both cases, SG allows flexibility of the other training-related parameters. For example, we can resume a training experiment and run it for more epochs:
+
+
+```shell
+> python train_from_recipe.py --config-name=cifar10_resnet experiment_name=cifar_experiment training_hyperparams.resume=True training_hyperparams.max_epochs=300
+...
+...
+> python train_from_recipe.py --config-name=cifar10_resnet experiment_name=cifar_experiment training_hyperparams.resume=True training_hyperparams.max_epochs=400
+```
+
+This means that such flexibility comes with its price: we must be aware of any change in parameters (by command line overrides or hard-coded changes inside the yaml file configurations) if we wish to resume training.
```

Comment:
please rephrase.
"However, this flexibility comes with a price"

Reply:
fixed https://github.com/Deci-AI/super-gradients/pull/703/commits/ad2036e85f0b857fe9123ca0407553f68a433422